frontend/src/hooks/queryBuilder/useQueryOperations.ts

```diff
@@ -0,0 +1,294 @@
+import {
+	initialAggregateAttribute,
+	initialQueryBuilderFormValues,
+	mapOfOperators,
+} from 'constants/queryBuilder';
+import { ITEMS } from 'container/NewDashboard/ComponentsSlider/menuItems';
+import { QueryProps } from 'container/QueryBuilder/components/Query/Query.interfaces';
+import { useQueryBuilderContext } from 'hooks/queryBuilder/useQueryBuilderContext';
+import { findDataTypeOfOperator } from 'lib/query/findDataTypeOfOperator';
+import React, {
+	ChangeEvent,
+	useCallback,
+	useEffect,
+	useMemo,
+	useState,
+} from 'react';
+import { BaseAutocompleteData } from 'types/api/queryBuilder/queryAutocompleteResponse';
+import {
+	Having,
+	IBuilderQueryForm,
+	TagFilter,
+} from 'types/api/queryBuilder/queryBuilderData';
+import {
+	DataSource,
+	ReduceOperators,
+	StringOperators,
+} from 'types/common/queryBuilder';
+
+type UseQueryOperationsParams = Pick<
+	QueryProps,
+	'index' | 'panelType' | 'query'
+>;
+
+type UseQueryOperations = (
+	params: UseQueryOperationsParams,
+) => {
+	isMetricsDataSource: boolean;
+	operators: string[];
+	handleChangeOperator: (value: string) => void;
+	handleChangeAggregatorAttribute: (value: BaseAutocompleteData) => void;
+	handleChangeDataSource: (newSource: DataSource) => void;
+	handleToggleDisableQuery: () => void;
+	handleChangeGroupByKeys: (values: BaseAutocompleteData[]) => void;
+	handleChangeQueryLegend: (e: ChangeEvent<HTMLInputElement>) => void;
+	handleChangeReduceTo: (value: ReduceOperators) => void;
+	handleChangeHavingFilter: (having: Having[]) => void;
+	handleDeleteQuery: () => void;
+	handleChangeOrderByKeys: (value: BaseAutocompleteData[]) => void;
+	handleChangeLimit: (value: number | null) => void;
+	handleChangeAggregateEvery: (value: number) => void;
+	handleChangeTagFilters: (value: TagFilter) => void;
+};
+
+export const useQueryOperations: UseQueryOperations = ({
+	query,
+	index,
+	panelType,
+}) => {
+	const { handleSetQueryData, removeEntityByIndex } = useQueryBuilderContext();
+	const [operators, setOperators] = useState<string[]>([]);
+
+	const { dataSource } = query;
+
+	const handleChangeOperator = useCallback(
```

Comment:
can you make a util function which return `updatedQueryData` based on different variables pointer

Reply:
why because let's say in future we wanna handle more cases at that time we are adding more cases and setting the value there

let me suggest you a better appraoch in which setting of queryData happen at one place only `handleSetQueryData`

Reply:
```
  const aggregateDataType: BaseAutocompleteData['dataType'] =
            query.aggregateAttribute.dataType;

        const typeOfValue = findDataTypeOfOperator(value);
        const shouldResetAggregateAttribute =
            (aggregateDataType === 'string' || aggregateDataType === 'bool') &&
            typeOfValue === 'number';

        let newQuery: IBuilderQueryForm = {
            ...query,
            aggregateOperator: value,
            having: [],
            groupBy: [],
            orderBy: [],
            limit: null,
            tagFilters: { items: [], op: 'AND' },
            ...(shouldResetAggregateAttribute
                ? { aggregateAttribute: initialAggregateAttribute }
                : {}),
        };

        handleSetQueryData(index, newQuery);
```